internal/core/metadata/init.go

```diff
@@ -146,6 +159,28 @@ func initializeConfiguration(useRegistry bool, useProfile string) (*Configuratio
 	return configuration, nil
 }
 
+func connectAndPollSecrets() error {
```

Comment:
It looks like this function and its contents are duplicated in other go files. Is it possible to relocate the function and reuse it rather than duplicating?

Reply:
Of course, but that's kind of the story of these init.go files that all do 90% of the same thing. We're definitely going deeper in tech debt here with this but I don't think it's invalid.

That being said, I'm open to starting to digging us out. It's just not obvious where it should go because the services are supposed to be "independent."

Reply:
Good point, this is probably a bigger concern regarding where to move all common logic.